metrics/test/features/dashboard/domain/usecases/receive_project_metrics_updates_test.dart

```diff
@@ -0,0 +1,242 @@
+import 'package:metrics/features/dashboard/domain/entities/core/project.dart';
+import 'package:metrics/features/dashboard/domain/entities/metrics/build_number_metric.dart';
+import 'package:metrics/features/dashboard/domain/entities/metrics/build_result_metric.dart';
+import 'package:metrics/features/dashboard/domain/entities/metrics/dashboard_project_metrics.dart';
+import 'package:metrics/features/dashboard/domain/entities/metrics/performance_metric.dart';
+import 'package:metrics/features/dashboard/domain/repositories/metrics_repository.dart';
+import 'package:metrics/features/dashboard/domain/usecases/parameters/project_id_param.dart';
+import 'package:metrics/features/dashboard/domain/usecases/receive_project_metrics_updates.dart';
+import 'package:metrics/util/date.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group("ReceiveProjectMetricUpdates", () {
+    const projectId = 'projectId';
+    final repository = MetricsRepositoryTestbed();
+    final receiveProjectMetricsUpdates =
+        ReceiveProjectMetricsUpdates(repository);
+
+    List<Build> builds;
+    Build lastBuild;
+    DashboardProjectMetrics projectMetrics;
+
+    setUpAll(() async {
+      builds = MetricsRepositoryTestbed.testBuilds;
+
+      projectMetrics =
+          await receiveProjectMetricsUpdates(const ProjectIdParam(projectId))
+              .first;
+
+      lastBuild = builds.first;
+    });
+
+    test("loads all fields in the performance metrics", () {
+      final performanceMetrics = projectMetrics.performanceMetrics;
+      final firstPerformanceMetric = performanceMetrics.buildsPerformance.last;
+
+      expect(
+        performanceMetrics.buildsPerformance.length,
+        builds.length,
+      );
+
+      expect(
+        firstPerformanceMetric.date,
+        lastBuild.startedAt,
+      );
+      expect(
+        firstPerformanceMetric.duration,
+        lastBuild.duration,
+      );
+    });
+
+    test("loads build number metric", () {
+      final timestamp = DateTime.now();
+      final buildsLoadingStartDate = timestamp
+          .subtract(ReceiveProjectMetricsUpdates.buildNumberLoadingPeriod)
+          .date;
+      final thisWeekBuilds = builds
+          .where((build) => build.startedAt.isAfter(buildsLoadingStartDate));
+
+      final totalNumberOfBuilds = thisWeekBuilds.length;
+      final buildNumberMetrics = projectMetrics.buildNumberMetrics;
+
+      expect(buildNumberMetrics.numberOfBuilds, totalNumberOfBuilds);
+    });
+
+    test('loads all fields in the build result metrics', () {
+      final buildResultMetrics = projectMetrics.buildResultMetrics;
+
+      final firstBuildResult = buildResultMetrics.buildResults.last;
+
+      expect(firstBuildResult.buildStatus, lastBuild.buildStatus);
+      expect(firstBuildResult.duration, lastBuild.duration);
+      expect(firstBuildResult.date, lastBuild.startedAt);
+      expect(firstBuildResult.url, lastBuild.url);
```

Comment:
We can't extend `Equatable` and compare the `firstBuildResult` and the `lastBuild` because the `firstBuildResult` is the `BuildResult` and the `lastBuild` is the `Build`. As for predicate matcher, I think it is better to compare each field separately because in case this test fails we will be notified on which filed it failed.